pegasus/sites.v3/code.org/public/pd-workshop-survey/splat.haml

```diff
@@ -0,0 +1,374 @@
+          = local_view :multi_select,
+            name: 'subjects_taught_ss[]',
+            label: 'What subjects have you taught? (select all that apply)'
+
+        .form-group{id: 'years-taught-cs-wrapper', style: 'display: none;'}
+          %label.control-label{for: 'years_taught_cs_i'}
+            For how many years have you taught Computer Science?
+            %span.form-required-field *
+          %input.form-control{name: 'years_taught_cs_i', type: 'text', maxlength: 2}
+
+    .main-section
+      .form-group.submit-btn
+        %button#btn-submit.btn.btn-primary{type: "submit"} Submit
+
+  #thanks{style: 'display: none;'}
+    %h1 Thank you.
+    %p Your submission was received.
+
+:javascript
+  $(document).ready(function () {
+    $('#pd-workshop-survey-form').find("select").selectize({
+      plugins: ['fast_click']
+    });
+
+    $('#pd-workshop-survey-form').submit(function (event) {
+      PdWorkshopSurveyFormSubmit(event);
+    });
+
+    $("input[name=consent_b]:radio").change(function () {
+      if ($("input[name=consent_b][value=1]").is(":checked")) {
+        $(".consent-only").show();
+      } else {
+        $(".consent-only").hide();
+      }
+    });
+
+    $("input[name=will_teach_b]:radio").change(function () {
+      if ($("input[name=will_teach_b][value=0]").is(":checked")) {
+        $("#will-teach-no-explain-wrapper").show();
+      } else {
+        $("#will-teach-no-explain-wrapper").hide();
+      }
+    });
+
+    $("select[name='reason_for_attending_ss[]']").change(function() {
+      if ($.inArray('Other', $(this).val()) > -1) {
+        $('#reason-for-attending-other-wrapper').show();
+      } else {
+        $('#reason-for-attending-other-wrapper').hide();
+      }
+    });
+
+    $("select[name='how_heard_ss[]']").change(function() {
+      if ($.inArray('Other', $(this).val()) > -1) {
+        $('#how-heard-other-wrapper').show();
+      } else {
+        $('#how-heard-other-wrapper').hide();
+      }
+    });
+
+    $("select[name='subjects_taught_ss[]']").change(function() {
+      if ($.inArray('Computer Science', $(this).val()) > -1) {
+        $('#years-taught-cs-wrapper').show();
+      } else {
+        $('#years-taught-cs-wrapper').hide();
+      }
+    });
+
+    $(".agree-scale").tooltip();
+
+
+  });
+
+  function processResponse(data)
+  {
+    $("#btn-submit").removeAttr('disabled');
+    $("#btn-submit").removeClass("button_disabled").addClass("button_enabled");
+    $('#pd-workshop-survey-form').hide();
+    $('#thanks').show();
+  }
+
+  function processError(data)
+  {
+    $('.has-error').removeClass('has-error');
+
+    var error_field_names = Object.keys(data.responseJSON);
+    var errors_count = error_field_names.length;
+
+    for (i = 0; i < errors_count; ++i) {
+      var error_field_name = error_field_names[i].replace(/_ss$/,'_ss[]');
+      var query = ".control-label[for='" + error_field_name + "']"
+      $(query).parents('.form-group').addClass('has-error');
```

Comment:
If I try submitting without filling anything out (except the first question), it will highlight most of the required questions in red. It doesn't highlight the last group of demographic questions because of the is_first_survey gate probably? Not a big deal, but would be nice if it was required in my case that these would also be highlighted the same.